hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -235,19 +235,21 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
   }
 
   @Override
-  public org.apache.iceberg.Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
+  public org.apache.iceberg.Table registerTable(TableIdentifier identifier,
+      String metadataFileLocation, boolean force) {
     Preconditions.checkArgument(isValidIdentifier(identifier), "Invalid identifier: %s", identifier);
 
-    // Throw an exception if this table already exists in the catalog.
-    if (tableExists(identifier)) {
+    // Throw an exception if this table already exists in the catalog and no force option.
+    if (tableExists(identifier) && !force) {
       throw new org.apache.iceberg.exceptions.AlreadyExistsException("Table already exists: %s", identifier);
     }
+    HiveTableOperations ops = (HiveTableOperations) newTableOps(identifier);
+    TableMetadata base = ops.current();
```

Comment:
Do we need to get the `base`? Passing the null should be fine, right?

Reply:
I'm not sure we should go through the operation.commit pathway here. Feels like what we are doing here is different than every other kind of commit. Really all we want to do is transactionally swap one parameter for another. I'm gonna think more about this tomorrow